### PR TITLE
Feature - add pill component

### PR DESCRIPTION
**Pull request by rikhilrai**
## Screenshots:

Screenshot 2020-11-22 at 16 04 30

## Checklist:

- [x] I have updated the CHANGELOG with a summary of my changes
- [x] I have updated the documentation accordingly

### CODE REVIEW

File: src/atoms/pill/theme.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import { IBoxTheme, ITextTheme } from '../../particles';
+
+export interface IPillThemeBase extends ThemeType {
+  text: ITextTheme;
+  background: IBoxTheme;
+}
+
+export interface IPillThemeState extends ThemeType {
+  default: IPillThemeBase;
+  // NOTE(rikhil): we can include hover, press, focus if we allow pills to be clicked
```
Comment (rikhilrai) [inline src/atoms/pill/theme.ts:11]: Do you want these to be clickable?

File: src/atoms/pill/component.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  iconGutter?: PaddingSize;
+  onClicked?(): void;
+}
+
+export const Pill = (props: IPillProps): React.ReactElement => {
+  const theme = useBuiltTheme('pills', props.variant, props.theme);
+  return (
+    <StyledPill
+      id={props.id}
+      className={getClassName(Pill.displayName, props.className, props.isFullWidth && 'fullWidth')}
```
Comment (rikhilrai) [inline src/atoms/pill/component.tsx:42]: fullWidth is not working for me for some reason. Despite having `fullWidth=false` the pill occupies the page width. Behaviour is different to button for example

File: src/atoms/pill/component.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import { IComponentProps, defaultComponentProps, themeToCss, useBuiltTheme } from '../..';
+import { IPillTheme } from './theme';
+import { IIconProps, Spacing, PaddingSize } from '../../particles';
+
+interface IStyledPillProps {
+  theme: IPillTheme;
+}
+
+const StyledPill = styled.div<IStyledPillProps>`
+  ${(props: IStyledPillProps): string => themeToCss(props.theme.normal.default.text)};
```
Comment (rikhilrai) [inline src/atoms/pill/component.tsx:14]: removing the not used ones. one moment

File: src/atoms/pill/component.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  align-items: center;
+  justify-content: center;
+  background-clip: border-box;
+  transition-duration: 0.3s;
+  &.fullWidth {
+    width: 100%;
+  }
+`;
+
+export interface IPillProps extends IComponentProps<IPillTheme> {
```
Comment (rikhilrai) [inline src/atoms/pill/component.tsx:28]: removing the not used ones. one moment

File: src/atoms/pill/theme.ts (hunk 1 above)
Comment (krishan711) [reply src/atoms/pill/theme.ts:11]: dont think so, they just contain information? I think if someone wants a clickable one they should use button. What do you think?

Comment (rikhilrai) [reply src/atoms/pill/theme.ts:11]: Comment inspired by what they have here https://material-ui.com/components/chips/.

**Comment by krishan711** [comment]
updated the theming slightly so they look less like buttons

File: src/atoms/pill/component.tsx (hunk 2 above)
Comment (krishan711) [reply src/atoms/pill/component.tsx:40]: had to use `display:inline-flex` i think cos html `<button />` components have this by default but `<div />`s are `display: block` by default

**Review by krishan711 — APPROVED** [review]

**Review by krishan711 — APPROVED** [review]

**Pull request merged**